test/server/filter_chain_benchmark_test.cc

```diff
@@ -0,0 +1,228 @@
+                             const std::vector<std::string>& application_protocols,
+                             const std::string& source_address, uint16_t source_port) {
+    auto res = std::make_unique<MockConnectionSocket>();
+
+    if (absl::StartsWith(destination_address, "/")) {
+      res->local_address_ = std::make_shared<Network::Address::PipeInstance>(destination_address);
+    } else {
+      res->local_address_ =
+          Network::Utility::parseInternetAddress(destination_address, destination_port);
+    }
+    if (absl::StartsWith(source_address, "/")) {
+      res->remote_address_ = std::make_shared<Network::Address::PipeInstance>(source_address);
+    } else {
+      res->remote_address_ = Network::Utility::parseInternetAddress(source_address, source_port);
+    }
+    res->server_name_ = server_name;
+    res->transport_protocol_ = transport_protocol;
+    res->application_protocols_ = application_protocols;
+    return res;
+  }
+
+  const Network::Address::InstanceConstSharedPtr& remoteAddress() const override {
+    return remote_address_;
+  }
+
+  const Network::Address::InstanceConstSharedPtr& localAddress() const override {
+    return local_address_;
+  }
+
+  absl::string_view detectedTransportProtocol() const override { return transport_protocol_; }
+
+  absl::string_view requestedServerName() const override { return server_name_; }
+  const std::vector<std::string>& requestedApplicationProtocols() const override {
+    return application_protocols_;
+  }
+
+  // Wont call
+  Network::IoHandle& ioHandle() override { return *io_handle_; }
+  const Network::IoHandle& ioHandle() const override { return *io_handle_; }
+
+  // Dummy method
+  void close() override {}
+  Network::Address::SocketType socketType() const override {
+    return Network::Address::SocketType::Stream;
+  }
+  void setLocalAddress(const Network::Address::InstanceConstSharedPtr&) override {}
+  void restoreLocalAddress(const Network::Address::InstanceConstSharedPtr&) override { return; }
+  void setRemoteAddress(const Network::Address::InstanceConstSharedPtr&) override {}
+  bool localAddressRestored() const override { return true; }
+  void setDetectedTransportProtocol(absl::string_view) override {}
+  void setRequestedApplicationProtocols(const std::vector<absl::string_view>&) override {}
+  void addOption(const OptionConstSharedPtr&) override {}
+  void addOptions(const OptionsSharedPtr&) override {}
+  const OptionsSharedPtr& options() const override { return options_; }
+  void setRequestedServerName(absl::string_view) override {}
+
+private:
+  Network::IoHandlePtr io_handle_;
+  OptionsSharedPtr options_;
+  Network::Address::InstanceConstSharedPtr local_address_;
+  Network::Address::InstanceConstSharedPtr remote_address_;
+  std::string server_name_;
+  std::string transport_protocol_;
+  std::vector<std::string> application_protocols_;
+};
+const std::string yaml_header = R"EOF(
```

Comment:
No exception in a benchmark test? A compile unit has _main_ and no other unit could refer these strings as they are all in anonymous namespace...

Will fix anyway to avoid burning the brain :)